ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -125,12 +142,28 @@ - (BOOL)acquireFromComponent:(CKComponent *)component
 {
   if (!_acquired && [component isMemberOfClass:_componentClass]) {
     _acquired = YES;
+    _acquiredComponent = component;
     return YES;
   } else {
     return NO;
   }
 }
 
+- (void)resolve
```

Comment:
Let me know what you think of this design. Acquisition is followed by resolution in the scope's destructor.